### PR TITLE
Deduplicate TokenDetailsService. Refactor TokenService -> TokensDetai…

### PR DESCRIPTION
**Pull request by gasiortomasz**

### CODE REVIEW

File: universal-login-wallet/src/integration/ethereum/TokensDetailsStore.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  async aggregateTokensDetails() {
+    for (const address of this.tokensAddresses) {
+      const details = await this.tokenDetailsService.getTokenDetails(address);
+      this.tokensDetails.push(details);
+    }
+  }
+
+  getTokenAddress(symbol: string) {
+    const token = this.tokensDetails.find(token => token.symbol === symbol);
+    return token ? token.address : undefined;
```
Comment (gasiortomasz) [inline universal-login-wallet/src/integration/ethereum/TokensDetailsStore.ts]: Will be null in next PR. #smallPRs

File: universal-login-commons/lib/integration/ethereum/TokenDetailsService.ts [hunk 2]
```diff
@@ -9,15 +10,21 @@ const tokenAbi = [
 export class TokenDetailsService {
   constructor(private provider: providers.Provider) {}
 
-  async getSymbol(tokenAddress: string) {
+  async getTokenDetails(tokenAddress: string): Promise<TokenDetails> {
+    const symbol = await this.getSymbol(tokenAddress);
+    const name = await this.getName(tokenAddress);
```
Comment (JustynaBroniszewska) [inline universal-login-commons/lib/integration/ethereum/TokenDetailsService.ts]: I am not sure if we use name or symbol separately, so I would inline `getSymbol` and `getName`.

File: universal-login-commons/lib/integration/ethereum/TokenDetailsService.ts [hunk 3]
```diff
@@ -9,15 +10,21 @@ const tokenAbi = [
...
+    return {address: tokenAddress, symbol, name};
+  }
+
+  async getSymbol(tokenAddress: string): Promise<string> {
     if (tokenAddress === ETHER_NATIVE_TOKEN.address) {
       return 'ETH';
```
Comment (JustynaBroniszewska) [inline universal-login-commons/lib/integration/ethereum/TokenDetailsService.ts]: `return ETHER_NATIVE_TOKEN.symbol`

File: universal-login-commons/lib/integration/ethereum/TokenDetailsService.ts [hunk 4]
```diff
@@ -9,15 +10,21 @@ const tokenAbi = [
...
     }
     const token = new Contract(tokenAddress, tokenAbi, this.provider);
     return token.symbol();
   }
 
-  async getName(tokenAddress: string) {
+  async getName(tokenAddress: string): Promise<string> {
     if (tokenAddress === ETHER_NATIVE_TOKEN.address) {
       return 'ether';
```
Comment (JustynaBroniszewska) [inline universal-login-commons/lib/integration/ethereum/TokenDetailsService.ts]: `return ETHER_NATIVE_TOKEN.name`

File: universal-login-wallet/test/integration/ui/CreateAccount.tsx [hunk 5]
```diff
@@ -23,7 +23,7 @@ describe('UI: Creation flow', () => {
         const [wallet] = await getWallets(createMockProvider());
         ({relayer, provider} = await setupSdk(wallet, '33113'));
         services = await createPreconfiguredServices(provider, relayer, [ETHER_NATIVE_TOKEN.address]);
-        services.tokenService.start();
+        services.tokensDetailsStore.aggregateTokensDetails();
```
Comment (JustynaBroniszewska) [inline universal-login-wallet/test/integration/ui/CreateAccount.tsx]: missing `await`

File: universal-login-wallet/test/integration/ui/connect.tsx [hunk 6]
```diff
@@ -27,7 +27,7 @@ describe('UI: Connection flow', () => {
     const [wallet] = getWallets(createMockProvider());
     ({relayer, provider} = await setupSdk(wallet, '33113'));
     services = await createPreconfiguredServices(provider, relayer, [ETHER_NATIVE_TOKEN.address]);
-    services.tokenService.start();
+    services.tokensDetailsStore.aggregateTokensDetails();
```
Comment (JustynaBroniszewska) [inline universal-login-wallet/test/integration/ui/connect.tsx]: missing `await`

File: universal-login-wallet/test/integration/helpers/setupUI.tsx [hunk 7]
```diff
@@ -13,7 +13,7 @@ export const setupUI = async (relayer: Relayer, tokenAddress?: string) => {
   const [wallet] = await getWallets(relayer.provider);
   const tokens = tokenAddress ? [tokenAddress, ETHER_NATIVE_TOKEN.address] : [ETHER_NATIVE_TOKEN.address];
   const services = await createPreconfiguredServices(relayer.provider, relayer, tokens);
-  services.tokenService.start();
+  services.tokensDetailsStore.aggregateTokensDetails();
```
Comment (JustynaBroniszewska) [inline universal-login-wallet/test/integration/helpers/setupUI.tsx]: missing `await`

File: universal-login-wallet/src/integration/ethereum/TokensDetailsStore.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import {TokenDetails, TokenDetailsService} from '@universal-login/commons';
+
+class TokensDetailsStore {
+  tokensDetails: TokenDetails[] = [];
+
+  constructor(private tokenDetailsService: TokenDetailsService, private tokensAddresses: string[]) {}
+
+  async aggregateTokensDetails() {
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/integration/ethereum/TokensDetailsStore.ts]: `aggregateTokensDetails`  -> `fetchTokensDetails`

Aggregate relates to operations that produce scalar from vector/matrix e.g. sum, avg, dev

File: universal-login-wallet/src/index.tsx [hunk 9]
```diff
@@ -10,7 +10,7 @@ import {CustomBrowserRouter} from './ui/react/CustomBrowserRouter';
 const config = getConfig();
 
 const services = createServices(config);
-services.tokenService.start();
+services.tokensDetailsStore.aggregateTokensDetails();
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/index.tsx]: We should aggregate all start methods into one `start()` methods of `service` object.

File: universal-login-wallet/test/integration/helpers/setupUI.tsx [hunk 10]
```diff
@@ -13,7 +13,7 @@ export const setupUI = async (relayer: Relayer, tokenAddress?: string) => {
   const [wallet] = await getWallets(relayer.provider);
   const tokens = tokenAddress ? [tokenAddress, ETHER_NATIVE_TOKEN.address] : [ETHER_NATIVE_TOKEN.address];
   const services = await createPreconfiguredServices(relayer.provider, relayer, tokens);
-  services.tokenService.start();
```
Comment (marekkirejczyk) [inline universal-login-wallet/test/integration/helpers/setupUI.tsx]: Duplication. We should extract `start` to `Services` class.

Comment (gasiortomasz) [reply universal-login-wallet/test/integration/helpers/setupUI.tsx]: This is test tho

**Pull request merged**